### DISCUSSION TITLE
Missing custom attributes after combining Docs using Docs.from_docs()

### DISCUSSION THREAD

**Discussion by thiippal** [explosion/spaCy#7479] [Help: Coding & Implementations] [answered]
Hi!

I'm currently developing new content for my [learning materials](https://applied-language-technology.readthedocs.io/). I'm adding discourse-level annotations from @amir-zeldes [GUM corpus](https://github.com/amir-zeldes/gum) to spaCy Doc and Span objects.

The corpus is distributed in CoNLL-U format, which I parse for the discourse annotations and combine them into SpanGroup objects:

```python
    # Next we create a spaCy SpanGroup object that holds all our Spans for elementary discourse
    # units. The 'doc' argument takes the Doc object that the Spans belong to as input, whereas
    # 'name' defines the key used to retrieve the spans from the 'spans' attribute of the Doc
    # object. Finally, 'spans' takes a list of Span objects to be included in the Span group.
    span_group = SpanGroup(doc=doc, name="edus", spans=edu_spans)
    
    # Assign the SpanGroup to the 'spans' attribute of the Doc object under the key 'edus'
    doc.spans['edus'] = span_group
    
    for i, span in enumerate(doc.spans['edus']):
        
        span._.edu_id = edu_ids[i]
        span._.target_id = target_ids[i]
        span._.relation = disc_rels[i]
        
    return doc
```

However, when I combine the resulting Doc objects into a single Doc using the `from_docs` method, all the custom annotations are no longer available: `docs.spans['edus']` returns a `KeyError`.

Based on #7317, I was under the assumption that the custom attributes would be preserved?

**Comment by adrianeboyd** [comment, marked as answer]
Hi, `doc.spans` is separate from the user data and it looks like it hasn't been included in `Doc.from_docs` yet.
